File: GymCRM.SchedulingAPI/Models/DTOs/TrainerWorkingHours.cs
namespace GymCRM.SchedulingAPI.Models.DTOs;

public class TrainerWorkingHours
{
    public Guid Id { get; set; }
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
    public DateTime DateCreatedUtc { get; set; }
    public DateTime DateModifiedUtc { get; set; }
    public TrainerDailyAvailability DailyAvailability { get; set; }
}

File: gymcrm.web/src/modules/scheduling/types/trainerWorkingHours.ts
// Mirrors GymCRM.SchedulingAPI.Models.DTOs.TrainerWorkingHours 1:1.
// StartTime/EndTime wire as "HH:mm" strings (see GymCRM.Shared/Utilities/TimeOnlyJsonConverter.cs),
// which is exactly what <input type="time"> produces and accepts.
export interface TrainerWorkingHours {
    id: string;
    startTime: string;
    endTime: string;
    dateCreatedUtc: string;
    dateModifiedUtc: string;
}
